awx/main/access.py

```diff
@@ -838,6 +861,9 @@ def can_change(self, obj, data):
         if data and 'name' in data:
             self.check_license(add_host_name=data['name'])
 
+            # Check the per-org limit
+            self.check_org_host_limit(data, add_host_name=data['name'])
```

Comment:
It's suspicious that this method is being called in `can_change` without `obj` being passed into it.

If I grok this correctly, we have to check when modifying the host, because the host might have previously been a duplicate name with another host, but then it changes to be a unique name, increasing the license impact. But it's also possible that it changes from a unique name to another unique name, and there's no way you can account for that without having `obj`. You might say that the same would apply to `self.check_license` above, and I would agree.

The org-specific check, however, has an additional problem that you won't be able to reliably determine the `inventory` field. It's possible that the user does a PATCH to change only the host `name`, and in that case IIRC, the `data` passed into `can_change` will only have the changing fields, meaning that you won't know the inventory, won't know the org, etc.

Reply:
It's definitely possible I got this wrong.  I'll take a closer look.

Reply:
I've changed it to a dictionary containing `org.inventory`, which if I'm following the reasoning correctly should be sufficient here.